### PR TITLE
Added mixing of euclid transforms

### PR DESCRIPTION
**Pull request by tomasmahlberg**

### CODE REVIEW

**Comment by tomasmahlberg** [comment]
@marcusnaslund

File: source/geometry/FloatEuclidTransform.ooc [hunk 1]
```diff
@@ -59,6 +59,11 @@ FloatEuclidTransform: cover {
 		}
 		result
 	}
+	mix: func (other: This, factor: Float) -> This {
+		This new(FloatVector3D mix(this translation, other translation, factor),
+			FloatRotation3D new(this rotation _quaternion sphericalLinearInterpolation(other rotation _quaternion, factor)),
+			Float mix(this scaling, other scaling, factor))
```
Comment (marcusnaslund) [inline source/geometry/FloatEuclidTransform.ooc]:
```ooc
This new(
	FloatVector3D mix(this translation, other translation, factor),
	FloatRotation3D new(this rotation _quaternion sphericalLinearInterpolation(other rotation _quaternion, factor)),
	Float mix(this scaling, other scaling, factor))
```
reads better, methinks.

File: test/geometry/FloatEuclidTransformTest.ooc [hunk 2]
```diff
@@ -80,6 +80,16 @@ FloatEuclidTransformTest: class extends Fixture {
...
 		})
+		this add("mix", func {
+			transform1 := FloatEuclidTransform new()
+			transform2 := FloatEuclidTransform new(FloatVector3D new(1.f, 2.f, 3.f), FloatRotation3D createFromEulerAngles(0.1f, 0.2f, 0.3f), 0.95f)
+			expect(transform1 mix(transform2, 0.f), is equal to(transform1))
+			expect(transform1 mix(transform2, 1.f), is equal to(transform2))
+			mixedTransform := transform1 mix(transform2, 0.5f)
+			expect(mixedTransform translation, is equal to(FloatVector3D new(0.5f, 1.f, 1.5f)))
+			expect(mixedTransform rotation, is equal to(FloatRotation3D new(Quaternion new(0.9958282709f, 0.0172071829f, 0.0532323383f, 0.0720868334f))))
+			expect(mixedTransform scaling, is equal to(0.975f))
```
Comment (marcusnaslund) [inline test/geometry/FloatEuclidTransformTest.ooc]: :+1:

File: source/geometry/FloatEuclidTransform.ooc (hunk 1 above)
Comment (tomasmahlberg) [inline source/geometry/FloatEuclidTransform.ooc]: Sure. No newline for the last parenthesis?

Comment (marcusnaslund) [inline source/geometry/FloatEuclidTransform.ooc]: Nah.

**Comment by tomasmahlberg** [comment]
Updated.

**Pull request merged**